src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+    private final int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int threshold = 12;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
```

Comment:
It's difficult to read and understand this method. Please try to simplify it. Try not to use nested "if"